include/substrait/type/Type.h

```diff
@@ -0,0 +1,701 @@
+
+  [[nodiscard]] const int& length() const {
+    return length_;
+  }
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const int length_;
+};
+
+class FixedChar : public TypeBase<TypeKind::kFixedChar> {
+ public:
+  explicit FixedChar(int length, bool nullable = false)
+      : TypeBase<TypeKind::kFixedChar>(nullable), length_(length){};
+
+  [[nodiscard]] const int& length() const {
+    return length_;
+  }
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const int length_;
+};
+
+class Varchar : public TypeBase<TypeKind::kVarchar> {
+ public:
+  explicit Varchar(int length, bool nullable = false)
+      : TypeBase<TypeKind::kVarchar>(nullable), length_(length){};
+
+  [[nodiscard]] const int& length() const {
+    return length_;
+  }
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const int length_;
+};
+
+class List : public TypeBase<TypeKind::kList> {
+ public:
+  explicit List(TypePtr elementType, bool nullable = false)
+      : TypeBase<TypeKind::kList>(nullable),
+        elementType_(std::move(elementType)){};
+
+  [[nodiscard]] const TypePtr& elementType() const {
+    return elementType_;
+  }
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const TypePtr elementType_;
+};
+
+class Struct : public TypeBase<TypeKind::kStruct> {
+ public:
+  explicit Struct(std::vector<TypePtr> types, bool nullable = false)
+      : TypeBase<TypeKind::kStruct>(nullable), children_(std::move(types)) {}
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] const std::vector<TypePtr>& children() const {
+    return children_;
+  }
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const std::vector<TypePtr> children_;
+};
+
+class Map : public TypeBase<TypeKind::kMap> {
+ public:
+  Map(TypePtr keyType, TypePtr valueType, bool nullable = false)
+      : TypeBase<TypeKind::kMap>(nullable),
+        keyType_(std::move(keyType)),
+        valueType_(std::move(valueType)) {}
+
+  [[nodiscard]] const TypePtr& keyType() const {
+    return keyType_;
+  }
+
+  [[nodiscard]] const TypePtr& valueType() const {
+    return valueType_;
+  }
+
+  [[nodiscard]] std::string signature() const override;
+
+  [[nodiscard]] bool isMatch(
+      const std::shared_ptr<const ParameterizedType>& type) const override;
+
+ private:
+  const TypePtr keyType_;
+  const TypePtr valueType_;
+};
+
+/// ParameterizedType represent a type in
```

Comment:
This sentence seems unfinished.  Also, this class is another class that does not appear to be adding value